tests/common/mod.rs

```diff
@@ -493,6 +493,18 @@ fn bytes_as_array() {
     assert_eq!(deserialized.as_ref(), src);
 }
 
+#[wasm_bindgen_test]
+fn bytes_from_mixed_array() {
+    // The string "xyz" cannot convert to a u8
+    let value = (100, "xyz".to_string(), true)
+        .serialize(&SERIALIZER)
+        .unwrap();
+    assert!(from_value::<serde_bytes::ByteBuf>(value).is_err());
```

Comment:
Nit: I'd recommend `.unwrap_err` instead. It has a bit better DX - not only it's slightly shorter, it also prints the unexpected value if Result for some reason succeeds (basically the direct opposite of `unwrap`).

Reply:
Heh, I wrote it that way at first but couldn't decided if that or the directness of the assert was better.  I just pushed a fix.  Also, FWIW I did check by hand that the error message was for the right error.